tests/functional/provider_client/test_onboarding_token_generation.py

```diff
@@ -0,0 +1,234 @@
+        odf_version = str(get_ocs_version_from_csv()).replace(".stable", "")
+        if "rhodf" in odf_version:
+            odf_version = get_odf_tag_from_redhat_catsrc()
+
+        ocp_version = get_latest_release_version()
+        nodepool_replicas = 2
+
+        create_hypershift_clusters(
+            cluster_names=[cluster_name],
+            ocp_version=ocp_version,
+            odf_version=odf_version,
+            setup_storage_client=True,
+            nodepool_replicas=nodepool_replicas,
+        )
+
+        log.info("Switch to the hosted cluster")
+        config.switch_to_cluster_by_name(cluster_name)
+
+        server = str(OCP().exec_oc_cmd("whoami --show-server", out_yaml_format=False))
+
+        assert (
+            cluster_name in server
+        ), f"Failed to switch to cluster '{cluster_name}' and fetch data"
+
+        log.info("Test create onboarding key")
+        HostedClients().download_hosted_clusters_kubeconfig_files()
+
+        assert len(
+            storage_clients.generate_client_onboarding_ticket_ui(storage_quota=8)
+        ), "Failed to get onboarding key"
+        assert HostedODF(cluster_name).get_storage_client_status() == "Connected"
+
+        log.info("Destroy hosted cluster")
+        assert destroy_hosted_cluster(cluster_name), "Failed to destroy hosted cluster"
+
+    @hci_provider_required
+    def test_onboarding_storageclient_from_hcp_cluster(
+        self, create_hypershift_clusters, destroy_hosted_cluster
+    ):
+        """
+        Test to verify that a new storageclient can be onboarded successfully from a hcp cluster
+        using the onboardin token generated from provider--storage--storageclients page
+        Steps:
+            1. Check ux-backend-server pod is up for provider cluster
+            2. Delete ux-backend-server pod
+            3. Check ux-backend-server pod is respinned.
+
+        """
+        log.info("Create hosted client")
+        cluster_name = get_random_hosted_cluster_name()
+        odf_version = str(get_ocs_version_from_csv()).replace(".stable", "")
+        if "rhodf" in odf_version:
+            odf_version = get_odf_tag_from_redhat_catsrc()
+
+        ocp_version = get_latest_release_version()
+        nodepool_replicas = 2
+
+        create_hypershift_clusters(
+            cluster_names=[cluster_name],
+            ocp_version=ocp_version,
+            odf_version=odf_version,
+            setup_storage_client=True,
+            nodepool_replicas=nodepool_replicas,
+        )
+
+        log.info("Switch to the hosted cluster")
+        config.switch_to_cluster_by_name(cluster_name)
+
+        server = str(OCP().exec_oc_cmd("whoami --show-server", out_yaml_format=False))
+
+        assert (
+            cluster_name in server
+        ), f"Failed to switch to cluster '{cluster_name}' and fetch data"
+
+        log.info("Test create onboarding key")
+        HostedClients().download_hosted_clusters_kubeconfig_files()
```

Comment:
`self.download_hosted_clusters_kubeconfig_files()` already happened at `HostedClients.do_deploy`, you can remove this line